src/igmp.c

```diff
@@ -271,6 +266,77 @@ void accept_igmp(int ifindex, size_t recvlen)
     }
 }
 
+static size_t build_ether_ipv4_mc(uint8_t *buf, const uint8_t *srcmac, uint32_t dst)
+{
+    struct ether_header *eh = (struct ether_header *)buf;
+
+    memset(eh, 0, sizeof(*eh));
+
+    memcpy(eh->ether_shost, srcmac, sizeof(eh->ether_shost));
+
```

Comment:
Minor nitpick -- you might want to use `ETHER_MAP_IP_MULTICAST(dst, eh->ether_dhost)` instead.  It's in `netinet/if_ether_h`.

Reply:
I was thinking "there is probably a macro for this, I should look it up", but I obviously dropped that ball. Thank you!